library/std/src/fs.rs

```diff
@@ -2980,6 +2980,19 @@ pub fn read_dir<P: AsRef<Path>>(path: P) -> io::Result<ReadDir> {
 ///
 /// [changes]: io#platform-specific-behavior
 ///
+/// # Symlinks
+/// On UNIX systems, it is impossible to manipulate the permission bits of a symlink itself[^1].
+/// Because of this, on those systems, this function will update the permission bits
+/// of the file pointed to by the symlink.
+///
+/// Note that this behavior can lead to privalage escalation vulnerabilites,
+/// where the ability to write a symlink in one directory allows you to
+/// cause the permissions of another directory to be modified.
```

Comment:
This is too vague, especially "write a symlink" (create?) and I don't think it's not specific to the target of the symlink being a directory.